### PR TITLE
Refactor Resource.ts typings

### PR DESCRIPTION
**Pull request by Peter-Maguire**
This PR clears up the typings for Resouce.ts by specifying the typing for the callback function for resource requests.

### CODE REVIEW

File: common/lib/client/resource.ts [hunk 1]
```diff
@@ -51,16 +45,19 @@ function unenvelope(callback: Function, format: Utils.Format | null) {
       }
     }
 
-    if (body.statusCode === undefined) {
+    if (!body) {
+      callback(new ErrorInfo('Body is missing', null));
```
Comment (owenpearson) [inline common/lib/client/resource.ts:49]: Would be good to add something to this error message to indicate where it's coming from, eg "unenvelope(): Body is missing"

File: common/lib/client/resource.ts [hunk 2]
```diff
@@ -216,11 +226,11 @@ class Resource {
             /* token has expired, so get a new one */
             rest.auth.authorize(null, null, function (err: Error) {
               if (err) {
-                callback(err);
+                callback(ErrorInfo.fromValues(err));
```
Comment (owenpearson) [inline common/lib/client/resource.ts:229]: Looks like `rest.auth.authorize` will only invoke with an `ErrorInfo` (sorry it should be typed better). You should be able to just change the callback arg type to `ErrorInfo` and not have to use this.

**Review by owenpearson — CHANGES_REQUESTED** [review]

File: common/lib/client/resource.ts (hunk 2 above)
Comment (Peter-Maguire) [reply common/lib/client/resource.ts]: Switched it to err: ErrorInfo

File: common/lib/client/resource.ts [hunk 3]
```diff
@@ -105,7 +107,7 @@ function logResponseHandler(callback: Function, method: HttpMethods, path: strin
         'Received; ' +
           urlFromPathAndParams(path, params) +
           '; Headers: ' +
-          paramString(headers) +
+          (headers ? paramString(headers) : 'None') +
```
[the remark this answers is not in the document]
Comment (Peter-Maguire) [reply common/lib/client/resource.ts]: Done

File: common/lib/client/resource.ts (hunk 1 above)
Comment (Peter-Maguire) [reply common/lib/client/resource.ts]: Done

**Review by owenpearson — APPROVED** [review]
LGTM, thanks

**Pull request merged**